waitfor.go

```diff
@@ -179,6 +180,26 @@ func GRPCWaiter(name string, target *TargetConfig) error {
 	return nil
 }
 
+// checkStatus checks if the given HTTP status code matches the pattern provided in the target configuration.
+func checkStatus(targetPattern string, code int) error {
+	if targetPattern != "200" {
+		// Safely compile and initialize  the regular expression pattern and verify if it's valid
+		pattern, err := regexp.Compile(targetPattern)
+		if err != nil {
+			return fmt.Errorf("invalid Regular Expression %v", err)
+		}
+		if !pattern.MatchString(strconv.Itoa(code)) {
+			return fmt.Errorf("%d status Code and %s regex didn't match", code, pattern.String())
+		}
+		// if the target is set to default "200", check if the status code is a successful status code
+	} else {
+		if !isSuccess(code) {
+			return fmt.Errorf("got %d ", code)
+		}
+	}
+	return nil
+}
```

Comment:
I've been looking at this a little more. I wonder if we make the default pattern `^2..$` then we can remove the `else` code block and the `isSuccess` method entirely.

What do you think?

Reply:
Yes I agree it would be much better that way.